### PR TITLE
Allow scenario deletion.

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: app/js/preferences/deleteScenario.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<form ng-submit="delete()">
+  <div class="grid-x">
+    <div class="cell">
+      <h3 id="delete-scenario-header"> Delete problem</h3>
```
Comment (DanielReid) [inline app/js/preferences/deleteScenario.html]: 'problem'

File: app/js/preferences/deleteScenarioController.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+define(['lodash'], function(_) {
+  var dependencies = [
+    '$scope',
+    '$modalInstance',
+    'scenario',
+    'callback'
+  ];
+  var EditScenarioTitleController = function(
```
Comment (DanielReid) [inline app/js/preferences/deleteScenarioController.js]: 'edit title'

File: test/endToEnd/scenarioTest.js [hunk 3]
```diff
@@ -106,7 +106,41 @@ function switchInSmaa(browser) {
...
+    .click('#create-new-scenario-button')
+    .pause(50) //pause needed to not get 'stale element' error
+    .waitForElementVisible('#delete-scenario-button')
+    .assert.containsText('#scenario-selector', scenarioTitle)
+    .click('#delete-scenario-button')
+    .waitForElementVisible('#delete-scenario-header')
+    .click('#delete-scenario-confirm-button')
+    .waitForElementVisible('#delete-scenario-disabled')
+    .assert.containsText('#scenario-selector', 'Default');
+}
```
Comment (DanielReid) [inline test/endToEnd/scenarioTest.js]: a test that the delete button is disabled when there's precisely 1 scenario would also be good I think; same for subproblem (forgot that yesterday)

Comment (chinook25) [reply test/endToEnd/scenarioTest.js]: That is being checked inside the tests (here line 124), I don't feel like a separate test is necessary.

**Pull request merged**